### DISCUSSION TITLE
uniform.py issue

### DISCUSSION THREAD

**Discussion by ghost** [PrincetonUniversity/athena#383] [Q&A]
Hi,

I'm trying to convert HDF5 data to a uniform grid using the `uniform.py` Python script.
The command is: 
```
python uniform.py ../sim.out2 ../uniformdata/sim.uniform 2 3 1
```
However, I got this error:

```
Traceback (most recent call last):
  File "uniform.py", line 208, in <module>
    main(**vars(args))
  File "uniform.py", line 76, in main
    for key, val in attributes:
  File ".../conda_envs/athenapp/lib/python3.8/site-packages/h5py/_hl/base.py", line 367, in __iter__
    for key in self._mapping:
  File ".../conda_envs/athenapp/lib/python3.8/site-packages/h5py/_hl/attrs.py", line 264, in __iter__
    cpl = self._id.get_create_plist()
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5g.pyx", line 408, in h5py.h5g.GroupID.get_create_plist
ValueError: Not a group (not a group)
```

I tried a few things but they didn't work. 
I wonder if it's a version thing or something need to be changed in `uniform.py`.
Thank you

**Comment by felker** [comment]
@sk117kn can you post `sim.out2` ?

**Reply by ghost** [reply]
The file is 20G so probably not a good idea.
I'll see if I can reproduce the issue with an example problem.

**Reply by ghost** [reply]
@felker @c-white 
I just cloned the latest code from this repository.
I ran the test problem "Turb".
I configured it with 
"python configure.py --prob=turb --eos isothermal -mpi -hdf5 --hdf5_path=$PHDF5_BASE -fft --fftw=$FFTW_BASE"
The test run finishes ok but the uniform.py issue is there.
Is it because the HDF5 version? (I've tried 1.10.1 and 1.12.0)
Another thing is I'm running the test in 2D like below:

"<mesh>
nx1        = 64         # Number of zones in X1-direction
x1min      = -0.5       # minimum value of X1
x1max      = 0.5        # maximum value of X1
ix1_bc     = periodic   # inner-X1 boundary flag
ox1_bc     = periodic   # outer-X1 boundary flag

nx2        = 64         # Number of zones in X2-direction
x2min      = -0.5       # minimum value of X2
x2max      = 0.5        # maximum value of X2
ix2_bc     = periodic   # inner-X2 boundary flag
ox2_bc     = periodic   # outer-X2 boundary flag

nx3        = 1          # Number of zones in X3-direction
x3min      = -0.5       # minimum value of X3
x3max      = 0.5        # maximum value of X3
ix3_bc     = periodic   # inner-X3 boundary flag
ox3_bc     = periodic   # outer-X3 boundary flag

refinement = none

<meshblock>
nx1        = 32
nx2        = 32
nx3        = 1"

I found on Google that some 2D issue with uniform.py happened before.
It looks like it was fixed.
Then I tried to run the same problem in 3D, basically making nx3 the same as the other two.
But the error message is still there.
I wonder if on your side you see the same thing or not?
Thank you.

**Comment by c-white** [comment]
Also, can you verify what version of the script you are using? I believe `for key, val in attributes:` has been on line 72 for the past 3 years, but the error message indicates line 76.

**Reply by ghost** [reply]
sorry it's because I added a few lines for printing
I just cloned the latest version and the issue is still there

**Comment by felker** [comment]
@sk117kn which version of `h5py` are you using? If it is 3.x.x, can you try 2.1.x?

**Reply by felker** [reply]
Also, please run `h5dump -A [file]` on the input files and post here.

**Comment by prasundhang** [comment]
Is this problem fixed? I am getting the same issue using uniform.py. FYI, I tried to use h5py versions: 2.9.0 and 3.6.0. Any solution will be really helpful.
